lib/ex_doc/language/erlang.ex

```diff
@@ -572,6 +572,14 @@ defmodule ExDoc.Language.Erlang do
             {{:., _, [module, name]}, _, args}, acc ->
               {{:t, [], args}, [{pp({module, name}), {module, name, length(args)}} | acc]}
 
+            ## remote type module.type/0
+            {:., _, [module, name]} = ast, acc ->
+              {ast, [{pp({module, name}), {module, name, 0}} | acc]}
+
+            ## remote type module.type/0
```

Comment:
This also fixes the case where the following would not be linked : 

```
          {:type, 41, :field_type,
            [
              {:atom, 41, :authorization_endpoint},
              {:remote_type, 41,
               [{:atom, 41, :uri_string}, {:atom, 41, :uri_string}, []]}
            ]}
```

The other clause handles the case where there's a union (i.e., `uri_string:string() | undefined`)